sympy/solvers/solveset.py

```diff
@@ -231,6 +235,26 @@ def _invert_complex(f, g_ys, symbol):
                                for g_y in g_ys if g_y != 0])
             return _invert_complex(f.args[0], exp_invs, symbol)
 
+    if isinstance(f, TrigonometricFunction):
```

Comment:
In `_invert_real` it is handled similar way.[ here ](https://github.com/sympy/sympy/blob/master/sympy/solvers/solveset.py#L175)
So I think it should be here only.